app/controllers/sections/consignee/ConsigneeExciseController.scala

```diff
@@ -0,0 +1,90 @@
+ * limitations under the License.
+ */
+
+package controllers.sections.consignee
+
+import controllers.BaseNavigationController
+import controllers.actions._
+import forms.sections.consignee.ConsigneeExciseFormProvider
+import models.requests.DataRequest
+import models.sections.info.DestinationType.TemporaryRegisteredConsignee
+import models.{Mode, NorthernIrelandRegisteredConsignor, NorthernIrelandWarehouseKeeper}
+import navigation.ConsigneeNavigator
+import pages.sections.consignee.ConsigneeExcisePage
+import pages.sections.info.DestinationTypePage
+import play.api.i18n.MessagesApi
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
+import services.UserAnswersService
+import views.html.sections.consignee.ConsigneeExciseView
+
+import javax.inject.Inject
+import scala.concurrent.Future
+
+class ConsigneeExciseController @Inject()(override val messagesApi: MessagesApi,
+                                          override val auth: AuthAction,
+                                          override val userAllowList: UserAllowListAction,
+                                          override val getData: DataRetrievalAction,
+                                          override val requireData: DataRequiredAction,
+                                          override val navigator: ConsigneeNavigator,
+                                          override val userAnswersService: UserAnswersService,
+                                          formProvider: ConsigneeExciseFormProvider,
+                                          val controllerComponents: MessagesControllerComponents,
+                                          view: ConsigneeExciseView
+                                         ) extends BaseNavigationController with AuthActionHelper {
+
+  def onPageLoad(ern: String, lrn: String, mode: Mode): Action[AnyContent] =
+    authorisedDataRequest(ern, lrn) {
+      implicit request =>
+        Ok(view(
+          fillForm(ConsigneeExcisePage, formProvider(isNorthernIrishTemporaryRegisteredConsignee)),
+          routes.ConsigneeExciseController.onSubmit(ern, lrn, mode),
+          isNorthernIrishTemporaryRegisteredConsignee
+        ))
+    }
+
+
+  def onSubmit(ern: String, lrn: String, mode: Mode): Action[AnyContent] =
+    authorisedDataRequestAsync(ern, lrn) {
+      implicit request =>
+        formProvider(isNorthernIrishTemporaryRegisteredConsignee).bindFromRequest().fold(
+          formWithErrors =>
+            Future.successful(
+              BadRequest(view(
+                formWithErrors,
+                routes.ConsigneeExciseController.onSubmit(ern, lrn, mode),
+                isNorthernIrishTemporaryRegisteredConsignee
+              ))
+            ),
+          exciseRegistrationNumber => {
+            userAnswersService.set(request.userAnswers.copy(ern = exciseRegistrationNumber)).flatMap {
+              _ =>
+                saveAndRedirect(ConsigneeExcisePage, exciseRegistrationNumber, mode)
+            }
```

Comment:
This doesn't need to change the ERN of the UserAnswers. It only needs to save the captured ERN against the `ConsigneeExcisePage`.

```suggestion
            saveAndRedirect(ConsigneeExcisePage, exciseRegistrationNumber, mode)
```